docs/en/operations/table_engines/mergetree.md

```diff
@@ -235,6 +235,13 @@ To check whether ClickHouse can use the index when running a query, use the sett
 
 The key for partitioning by month allows reading only those data blocks which contain dates from the proper range. In this case, the data block may contain data for many dates (up to an entire month). Within a block, data is sorted by primary key, which might not contain the date as the first column. Because of this, using a query with only a date condition that does not specify the primary key prefix will cause more data to be read than for a single date.
 
+### Use of Index for Partially-Monotonic Primary Keys
+
+Consider, for example, the days of the month. They are the monotonic sequence inside one month, but they are not monotonic for a more extended period. This is the partially-monotonic sequence. If a user creates the table with such partially-monotonic primary key, ClickHouse creates a sparse index as usual. When a user selects data from such a table, ClickHouse analyzes query conditions. If the user wants to get data between two marks of the index and both this marks are within one month, ClickHouse can use the index in this particular case because it can calculate the distance between parameters of query and index marks.
```

Comment:
we'd better start with formal explanation on what the "partially-monotonic sequence" is, since it might be hard for reader to generalize just from example

Reply:
I have added the link to monotonic functions in wiki, there are a formal description of monotonic sequence and some examples of monotonic and non-monotonic functions.